src/Microsoft.VisualStudio.Editors/SettingsDesigner/TypePickerDialog.vb

```diff
@@ -446,15 +447,19 @@ Namespace Microsoft.VisualStudio.Editors.SettingsDesigner
                 Dim assemblyImage As Bitmap = GetManifestBitmapTransparent("assembly.bmp", Color.FromArgb(255, 0, 255))
                 Dim namespaceImage As Bitmap = GetManifestBitmapTransparent("namespace.bmp", Color.FromArgb(255, 0, 255))
                 Dim objectImage As Bitmap = GetManifestBitmapTransparent("object.bmp", Color.FromArgb(255, 0, 255))
+                'Scale the images for per monitor DPI
+                Handle.LogicalToDeviceUnits(assemblyImage.Width)
+                Handle.LogicalToDeviceUnits(assemblyImage.Height)
+                Handle.LogicalToDeviceUnits(namespaceImage.Width)
+                Handle.LogicalToDeviceUnits(namespaceImage.Height)
+                Handle.LogicalToDeviceUnits(objectImage.Width)
+                Handle.LogicalToDeviceUnits(objectImage.Height)
 
                 Dim treeViewIcons As ImageList = New ImageList()
                 treeViewIcons.Images.Add(assemblyImage)
                 treeViewIcons.Images.Add(namespaceImage)
                 treeViewIcons.Images.Add(objectImage)
 
-                'Scale the imagelist for High DPI
-                DpiHelper.LogicalToDeviceUnits(treeViewIcons)
```

Comment:
@etbyrd I suspect this method was resizing the images within the `ImageList`. I'm not sure if image monikers can be used with a regular tree view. If not, you may need to find or write an equivalent helper method for `ImageList`. Note that `ImageList` removes all its contents when you change its size, so if you're going to specify a size on the `ImageList` you must do so before adding images.

I wrote some similar code for GitExtensions a while back which may be helpful:

https://github.com/gitextensions/gitextensions/blob/c4764500400604a18869d6d3b78eda9271727210/GitUI/BranchTreePanel/RepoObjectsTree.cs#L84-L129

(Using that project's [`DpiUtil`](https://github.com/gitextensions/gitextensions/blob/master/GitExtUtils/GitUI/DpiUtil.cs) class)

Reply:
The expectation is that we move to image monikers to avoid the bitmap stretching.

Reply:
Sounds good. I don't know how this is done with a WinForms `TreeView`. AFAIK you can only set icons using `ImageList`.

Reply:
They draw using vectors, and then give a bitmap of that size, rather than resizing a fixed bitmap.

Reply:
I would presume there are hand-drawn versions for 96dpi as scaling vector art to 16/24/32px usually looks terrible ([see](https://www.pushing-pixels.org/2011/11/04/about-those-vector-icons.html)). Regardless, I agree image monikers are the better solution.

Do you know what API can be used to create an `ImageList` from a list of image monikers?

If this has to support PMA then would the `ImageList` need to be recreated on changing monitors?

Reply:
See my comment here: https://github.com/dotnet/project-system/pull/4766#discussion_r279569328. All of our WinForms code should be System-Aware and not PMA, so we don't need to react to changing monitors.